test/freenet/client/async/SplitFileInserterStorageTest.java

```diff
@@ -1365,8 +1365,11 @@ public void testCancelAltCrossSegment() throws IOException, InsertException, Mis
         Random r = new Random(0xb395f44d);
         testCancelAlt(r, CHKBlock.DATA_LENGTH*128*21);
     }
-    
+
     private void testCancelAlt(Random r, long size) throws IOException, InsertException {
+        // FIXME tricky to wait for "all threads are in pread()", when # threads != # segments.
+        // So just set max threads to 1 (only affects this test).
+        memoryLimitedJobRunner.setMaxThreads(1);
```

Comment:
The current code only calls it with 1 segment and with 21 segments. In both cases we can only run one FEC job at once because the memory limit is 9MB. However if we were to call it with a size of 7MB or so, it might try to do 2 jobs at once. In which case the test would fail, because we are assuming that *all the running FEC threads* are in pread().